File: hw/aymacana/u3/hw28Principles/CustomerPY/views/contact_view.py
import sys
from typing import Optional, Dict, Any
from models.contact_model import ContactType
from controllers.contact_controller import ContactController

class ContactView:
    def __init__(self, controller: ContactController):
        self.controller = controller
    
    def display_menu(self) -> None:
        print("\n" + "="*50)
        print("CONTACT MANAGEMENT SYSTEM")
        print("="*50)
        print("1. View all contacts")
        print("2. Search contact by ID")
        print("3. Exit")
        print("="*50)
    
    def get_user_choice(self) -> str:
        while True:
            try:
                choice = input("\nSelect an option (1-3): ").strip()
                if choice in ['1', '2', '3']:
                    return choice
                else:
                    print("Please select a valid option (1-3)")
            except KeyboardInterrupt:
                print("\nGoodbye!")
                sys.exit(0)
            except Exception as e:
                print(f"Error: {e}")
    
    def get_contact_id_input(self) -> Optional[str]:
        try:
            contact_id = input("\nEnter contact ID: ").strip()
            if not contact_id:
                print("ID not provided")
                return None
            return contact_id
        except KeyboardInterrupt:
            print("\nGoodbye!")
            sys.exit(0)
    
    def run(self) -> None:
        print("Starting Contact Management System...")
        
        while True:
            try:
                self.display_menu()
                choice = self.get_user_choice()
                
                if choice == '1':
                    self.controller.display_all_contacts()
                
                elif choice == '2':
                    contact_id = self.get_contact_id_input()
                    if contact_id:
                        self.controller.display_contact_by_id(contact_id)
                
                elif choice == '3':
                    print("\nThank you!!")
                    break
                
                input("\nPress Enter to continue...")
                
            except KeyboardInterrupt:
                print("\nGoodbye!")
                break
            except Exception as e:
                print(f"\nUnexpected error: {e}")
                input("\nPress Enter to continue...")

File: hw/aymacana/u3/hw28Principles/CustomersJS/views/ContactView.js
const readline = require('readline');

class ContactView {
    constructor(contactController) {
        this.contactController = contactController;
        this.rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });
    }

    displayWelcome() {
        console.clear();
        console.log('='.repeat(60));
        console.log('CONTACT MANAGEMENT SYSTEM');
        console.log('='.repeat(60));
        console.log('Database: oop');
        console.log('Collection: Customers');
        console.log('='.repeat(60) + '\n');
    }

    displayMenu() {
        console.log('\nMAIN MENU:');
        console.log('1. View all contacts');
        console.log('2. Search contact by ID');
        console.log('3. Exit');
    }

    async getInput(prompt) {
        return new Promise((resolve) => {
            this.rl.question(prompt, (answer) => {
                resolve(answer);
            });
        });
    }

    async handleSearchById() {
        const id = await this.getInput('Enter contact ID: ');
        if (id && !isNaN(id)) {
            await this.contactController.displayContactById(parseInt(id));
        } else {
            console.log('Invalid ID');
        }
    }

    async pressToContinue() {
        await this.getInput('\nPress Enter to continue...');
    }

    async run() {
        this.displayWelcome();
        
        let running = true;
        
        while (running) {
            try {
                this.displayMenu();
                const choice = await this.getInput('\nSelect option (1-3): ');
                
                switch (choice) {
                    case '1':
                        await this.contactController.displayAllContacts();
                        break;
                    case '2':
                        await this.handleSearchById();
                        break;
                    case '3':
                        running = false;
                        break;
                    default:
                        console.log('Invalid option');
                }
                
                if (running && (choice === '1' || choice === '2')) {
                    await this.pressToContinue();
                    console.log('\n');
                }
                
            } catch (error) {
                console.error(`Error: ${error.message}`);
                await this.pressToContinue();
            }
        }
        
        this.rl.close();
        console.log('\nThank you for using Contact Management System!');
    }
}

module.exports = ContactView;